### PR TITLE
Bc/imshowhs

### PR DESCRIPTION
**Pull request by BCampforts**
### Description
Fixed some bugs in the *imshowhs_grid* plotting function, also addressing issue #1484 
Fix 1: Before, *imshowhs_grid* inverted the plot in north-south directions, when a no-data drape was provided. This issue is solved.
Fix 2: Axis tick marks were slightly off. This issue is fixed.
Fix 3: The hillshade base layer was not plotting data from the rows and columns adjacent to the boundary nodes. This issue is solved.
Fix 4: Boundary nodes were not indicated if requested. This issue is now solved. For now, boundary nodes can be displayed in white (``color_for_closed = 'white' ``) or not at all (``color_for_closed = None ``)

### Checklist - did you ...

- [x] Add a [news fragment](https://landlab.readthedocs.io/en/master/development/contribution/index.html#news-entries) file entry if necessary?
- [x] Add / update tests if necessary?
- [x] Add new / update outdated documentation?
- [x] All tests have passed?
- [x] Formatted code with black?
- [x] Removed lint reported by flake8?
- [x] Sucessful documentation built? (if documentation added or modified)

### CODE REVIEW

**Comment by BCampforts** [comment]
Hi @mcflugen, I do not manage to build the docs for this. Not sure how to resolve it (see checks; Documentation)

File: landlab/plot/imshowhs.py [hunk 1]
```diff
@@ -454,6 +456,8 @@ def _imshowhs_grid_values(
     cmap = plt.get_cmap(cmap)
 
     if color_for_closed is not None:
+        if color_for_closed != "white":
```
Comment (mcflugen) [inline landlab/plot/imshowhs.py:459]: Why limit *color_for_closed* to only *"white"* or *None*? It seems like an [matplotlib color](https://matplotlib.org/stable/tutorials/colors/colors.html#sphx-glr-tutorials-colors-colors-py) should work.

Comment (BCampforts) [reply landlab/plot/imshowhs.py:459]: Good point, to plot the hillshade with an extra layer, I first create an RGB array and next change the transparency before plotting using pyplot's imshow. Didn't know about a quick fix to make this working with masks. 
See: 
`rgb[:, :, 3][values.mask] = 0`
` ima = ax1.imshow(rgb, extent=extent, **kwds)`

**Comment by mcflugen** [comment]
> Hi @mcflugen, I do not manage to build the docs for this. Not sure how to resolve it (see checks; Documentation)

The problem was that your news fragment file was not properly-formatted reStructuredText. I think I've fixed it now, though. In addition, since you fixed multiple bugs, you should include multiple news fragments (one for each bug fix). I've done this for you but you should probably read them over to make sure they make sense.

I just realized that each of the news fragments refer to #1484. If some of those bug fixes don't relate to that issue, we should change their file name to point to their issue/pull request.

File: landlab/plot/imshowhs.py (hunk 1 above)
Comment (mcflugen) [reply landlab/plot/imshowhs.py:459]: I think [to_rgba](https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.to_rgba.html?highlight=to_rgba) should do what you want.
```python
>>> from matplotlib.colors import to_rgba
>>> to_rgba("w")
(1.0, 1.0, 1.0, 1)
>>> to_rgba("w", alpha=0.0)
(1.0, 1.0, 1.0, 0.0)
>>> to_rgba("c")
(0.0, 0.0, 0.0, 1)
>>> to_rgba("k")
(0.0, 0.75, 0.75, 1)
```

Comment (BCampforts) [reply landlab/plot/imshowhs.py:459]: Yup, that's it!

**Comment by mcflugen** [comment]
@BCampforts I made some edits to the *docstrings* of the *imshowhs* functions. Could you please have a look?

I just copied the *docstring* for *imshowhs_grid*  to *imshowhs_grid_at_node*, is that ok? That seems to be what you had done. If they are the same, that indicates to me there should probably be just one function.

**Review by mcflugen — APPROVED** [review]

**Pull request merged**